### PR TITLE
Update "How to Write a Checker" docs

### PR DESCRIPTION
**Pull request by jpy-git**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :scroll: Docs          |

## Description

I just created my first AST checker in #5938 following the docs and wanted to make a few small tweaks to help improve them further for future contributors 😄

### CODE REVIEW

File: doc/how_tos/custom_checkers.rst [hunk 1]
```diff
@@ -37,30 +37,36 @@ Firstly we will need to fill in some required boilerplate:
...
   import astroid
   from astroid import nodes
+  from typing import TYPE_CHECKING
 
   from pylint.checkers import BaseChecker
   from pylint.interfaces import IAstroidChecker
-  from pylint.lint import PyLinter
+
+  if TYPE_CHECKING:
+      from pylint.lint import PyLinter
```
Comment (jpy-git) [inline doc/how_tos/custom_checkers.rst:46]: We were lazy loading this in the `def register(...` section so let's just put this once here.

File: doc/how_tos/custom_checkers.rst [hunk 2]
```diff
@@ -79,20 +85,19 @@ So far we have defined the following required components of our checker:
...
      prefixed with a **message category**.
      There are multiple message categories,
      these being ``C``, ``W``, ``E``, ``F``, ``R``,
      standing for ``Convention``, ``Warning``, ``Error``, ``Fatal`` and ``Refactoring``.
-     The rest of the 5 digits should not conflict with existing checkers
-     and they should be consistent across the checker.
-     For instance,
-     the first two digits should not be different across the checker.
+     The 4 digits should not conflict with existing checkers
+     and the first 2 digits should consistent across the checker.
```
Comment (jpy-git) [inline doc/how_tos/custom_checkers.rst:100]: There's only 4 digits, and this is the same thing but more succinct

File: doc/how_tos/custom_checkers.rst [hunk 3]
```diff
@@ -107,9 +112,10 @@ The options list defines any user configurable options.
 It has the following format::
 
     options = (
-        'option-symbol': {'argparse-like-kwarg': 'value'},
+        ("option-symbol", {"argparse-like-kwarg": "value"}),
```
Comment (jpy-git) [inline doc/how_tos/custom_checkers.rst:115]: this should be a tuple of tuples

File: doc/how_tos/custom_checkers.rst [hunk 4]
```diff
@@ -119,8 +125,8 @@ Next we'll track when we enter and leave a function.
 
 .. code-block:: python
 
-  def __init__(self, linter: PyLinter =None) -> None:
-      super(UniqueReturnChecker, self).__init__(linter)
+  def __init__(self, linter: Optional["PyLinter"] = None) -> None:
```
Comment (jpy-git) [inline doc/how_tos/custom_checkers.rst:128]: fixed the typing here

**Review by jpy-git — COMMENTED** [review]
Ran black over these examples to make them consistent with the rest of the codebase and I've called out the other small changes 😄

File: doc/how_tos/custom_checkers.rst [hunk 5]
```diff
@@ -183,13 +189,14 @@ Now we know how to use the astroid node, we can implement our check.
...
-             self.add_message(
-                 'non-unique-returns', node=node,
-             )
+          if node.value.value == other_return.value.value and not (
+              self.config.ignore_ints and node.value.pytype() == int
+          ):
+              self.add_message(
+                  "non-unique-returns",
+                  node=node,
+              )
```
Comment (Pierre-Sassoulas) [inline doc/how_tos/custom_checkers.rst:199]:
```suggestion
              self.add_message("non-unique-returns", node=node)
```

**Review by Pierre-Sassoulas — APPROVED** [review]
Thank you for the cleanup !

**Pull request merged**